train_rl4lm.py

```diff
@@ -106,29 +99,52 @@ def make_model_config(path):
     can be loaded by AutoModel.from_pretrained. 
     This means we need to instantiate a model, and call .save_pretrained
     rather than saving the config directly '''
-    from transformers import GPT2Config
+    from transformers import GPT2Config, GPT2LMHeadModel
 
     config = GPT2Config(
         activation_function='gelu_new',
         n_head=4,
         n_layer=2,
+        n_ctx = 32,
         hidden_size=128,
-        n_positions=100,  # upper bound on max length of input
+        n_positions=32,  # upper bound on max length of input
+        vocab_size = 50000
     )
-    model = AutoModelForCausalLM.from_config(
-      config
-    )
-    assert type(model) == transformers.models.gpt2.modeling_gpt2.GPT2LMHeadModel, \
-        "Model is not of type GPT2LMHeadModel, is of type {}".format(type(model))
+    
+    import torch.nn as nn
+
+    class RandomGPT2(GPT2LMHeadModel):
+        def _init_weights(self, module):
+            if isinstance(module, (nn.Linear, nn.Embedding, nn.LayerNorm)):
+                module.weight.data.normal_(mean=0.0, std=0.02)
+                if isinstance(module, (nn.Linear, nn.LayerNorm)) and module.bias is not None:
+                    module.bias.data.zero_()
+            elif isinstance(module, nn.MultiheadAttention):
+                module.in_proj_weight.data.normal_(mean=0.0, std=0.02)
+
+
+    model = RandomGPT2(config)
+    model.save_pretrained('random_gpt2')
+
+    # print(f"model is {model}")
+
+    
+    # assert type(model) == transformers.models.gpt2.modeling_gpt2.GPT2LMHeadModel, \
+    #     "Model is not of type GPT2LMHeadModel, is of type {}".format(type(model))
     print('created model', model)
```

Comment:
This is not necessary, as RL4LMs doesn't use this custom class, it just uses the AutoModel. In fact part of the test is to make sure that AutoClass loads the correct subtype